presto-main/src/main/java/com/facebook/presto/cost/SimpleStatsRule.java

```diff
@@ -36,8 +36,23 @@ protected SimpleStatsRule(StatsNormalizer normalizer)
     @Override
     public final Optional<PlanNodeStatsEstimate> calculate(T node, StatsProvider sourceStats, Lookup lookup, Session session, TypeProvider types)
     {
-        return doCalculate(node, sourceStats, lookup, session, types)
+        Optional<PlanNodeStatsEstimate> planNodeStatsEstimate = doCalculate(node, sourceStats, lookup, session, types)
                 .map(estimate -> normalizer.normalize(estimate, node.getOutputVariables()));
+        if (node.getSources().isEmpty()) {
+            // dont do the confident check for tablescan stats
+            return planNodeStatsEstimate;
+        }
+        boolean confident = sourceStats.getStats(node.getSources().get(0)).isConfident();
+        for (PlanNode source : node.getSources()) {
```

Comment:
Confidence level should not only depends on the source inputs, for example, EnforceSingleRowNode node should always be confident that the output is one single row. We need to exclude these rules from this check.

Reply:
I see maybe better to add this to an abstract method and override it in those rules